src/rdkafka_admin.c

```diff
@@ -1647,6 +1647,55 @@ rd_kafka_error_t *rd_kafka_AdminOptions_set_match_consumer_group_states(
         return !err ? NULL : rd_kafka_error_new(err, "%s", errstr);
 }
 
+rd_kafka_error_t *rd_kafka_AdminOptions_set_match_consumer_group_types(
+    rd_kafka_AdminOptions_t *options,
+    const rd_kafka_consumer_group_type_t *consumer_group_types,
+    size_t consumer_group_types_cnt) {
+        size_t i;
+        char errstr[512];
+        rd_kafka_resp_err_t err;
+        rd_list_t *types_list  = rd_list_new(0, NULL);
+        uint64_t types_bitmask = 0;
+
+        rd_list_init_int32(types_list, consumer_group_types_cnt);
+
+        if (RD_KAFKA_CONSUMER_GROUP_TYPE__CNT >= 64) {
+                rd_assert("BUG: cannot handle types with a bitmask anymore");
+        }
+
+        for (i = 0; i < consumer_group_types_cnt; i++) {
+                uint64_t type_bit;
+                rd_kafka_consumer_group_type_t type = consumer_group_types[i];
+
+                if (type <= RD_KAFKA_CONSUMER_GROUP_TYPE_UNKNOWN ||
+                    type >= RD_KAFKA_CONSUMER_GROUP_TYPE__CNT) {
+                        rd_list_destroy(types_list);
+                        return rd_kafka_error_new(
+                            RD_KAFKA_RESP_ERR__INVALID_ARG,
+                            "Only a valid and known type is allowed");
+                }
```

Comment:
Can we again split this? I told to split in 2 parts. 1 for UNKNOWN and other for any other invalid value. For better and more specific error messages.


```
               if (group_type < RD_KAFKA_CONSUMER_GROUP_TYPE_UNKNOWN ||
                    group_type >= RD_KAFKA_CONSUMER_GROUP_TYPE__CNT) {
                        rd_list_destroy(types_list);
                        return rd_kafka_error_new(
                            RD_KAFKA_RESP_ERR__INVALID_ARG,
                            "Only a valid group type is allowed");
                } else if (group_type == RD_KAFKA_CONSUMER_GROUP_TYPE_UNKNOWN) {
                        rd_list_destroy(types_list);
                        return rd_kafka_error_new(
                            RD_KAFKA_RESP_ERR__INVALID_ARG,
                            "UNKNOWN group type is not allowed");
                }
```